src/network/network_osm_reader.cpp

```diff
@@ -0,0 +1,361 @@
+// network_osm_reader.cpp -- function to read OSM data as network
+
+#include "network/network.hpp"
+#include "util/debug.hpp"
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include <osmium/handler.hpp>
+#include <osmium/io/any_input.hpp>
+#include <osmium/osm/node.hpp>
+#include <osmium/osm/way.hpp>
+#include <osmium/visitor.hpp>
+
+
+using namespace FMM;
+using namespace FMM::CORE;
+using namespace FMM::MM;
+using namespace FMM::NETWORK;
+
+
+
+/** Helper class to filter out OSM ways that are not roads */
+class way_filter {
```

Comment:
To keep it consistent with the styles, I will rename the class to `WayFilter`.

Reply:
Sure, I agree